src/game/mario_actions_cutscene.c

```diff
@@ -183,6 +230,192 @@ Gfx *geo_switch_peach_eyes(s32 callContext, struct GraphNode *node, UNUSED s32 c
+                    || gPrevLevel == LEVEL_BOWSER_3) {
+                    cutscene = CUTSCENE_EXIT_BOWSER_SUCC;
+                } else {
+                    cutscene = CUTSCENE_EXIT_SPECIAL_SUCC;
+                }
+                break;
+            case ACT_SPECIAL_DEATH_EXIT:
+                if (gPrevLevel == LEVEL_BOWSER_1 || gPrevLevel == LEVEL_BOWSER_2
+                    || gPrevLevel == LEVEL_BOWSER_3) {
+                    cutscene = CUTSCENE_EXIT_BOWSER_DEATH;
+                } else {
+                    cutscene = CUTSCENE_NONPAINTING_DEATH;
+                }
+                break;
+            case ACT_ENTERING_STAR_DOOR:
+                if (c->doorStatus == DOOR_DEFAULT) {
+                    cutscene = CUTSCENE_SLIDING_DOORS_OPEN;
+                } else {
+                    cutscene = CUTSCENE_DOOR_PULL_MODE;
+                }
+                break;
+            case ACT_UNLOCKING_KEY_DOOR:
+                cutscene = CUTSCENE_UNLOCK_KEY_DOOR;
+                break;
+            case ACT_WATER_DEATH:
+                cutscene = CUTSCENE_WATER_DEATH;
+                break;
+            case ACT_DEATH_ON_BACK:
+                cutscene = CUTSCENE_DEATH_ON_BACK;
+                break;
+            case ACT_DEATH_ON_STOMACH:
+                cutscene = CUTSCENE_DEATH_ON_STOMACH;
+                break;
+            case ACT_STANDING_DEATH:
+                cutscene = CUTSCENE_STANDING_DEATH;
+                break;
+            case ACT_SUFFOCATION:
+                cutscene = CUTSCENE_SUFFOCATION_DEATH;
+                break;
+            case ACT_QUICKSAND_DEATH:
+                cutscene = CUTSCENE_QUICKSAND_DEATH;
+                break;
+            case ACT_ELECTROCUTION:
+                cutscene = CUTSCENE_STANDING_DEATH;
+                break;
+            case ACT_STAR_DANCE_EXIT:
+                cutscene = determine_dance_cutscene(c);
+                break;
+            case ACT_STAR_DANCE_WATER:
+                cutscene = determine_dance_cutscene(c);
+                break;
+            case ACT_STAR_DANCE_NO_EXIT:
+                cutscene = CUTSCENE_DANCE_DEFAULT;
+                break;
+        }
+        switch (sMarioCamState->cameraEvent) {
+            case CAM_EVENT_START_INTRO:
+                cutscene = CUTSCENE_INTRO_PEACH;
+                break;
+            case CAM_EVENT_START_GRAND_STAR:
+                cutscene = CUTSCENE_GRAND_STAR;
+                break;
+            case CAM_EVENT_START_ENDING:
+                cutscene = CUTSCENE_ENDING;
+                break;
+            case CAM_EVENT_START_END_WAVING:
+                cutscene = CUTSCENE_END_WAVING;
+                break;
+            case CAM_EVENT_START_CREDITS:
+                cutscene = CUTSCENE_CREDITS;
+                break;
+        }
+    }
+    //! doorStatus is reset every frame. CameraTriggers need to constantly set doorStatus
+    c->doorStatus = DOOR_DEFAULT;
+
+    return cutscene;
+}
+
```

Comment:
is all this crap just moved here from camera.c?

Reply:
This seemed like a better place for it within the existing files than camera.c itself, though if a custom file can hold all this, then I can do that

Reply:
Yeah thats fine, just was unclear